packages/ilmomasiina-models/src/schema/auditLog/index.ts

```diff
@@ -0,0 +1,65 @@
+import { Static, Type } from '@sinclair/typebox';
+
+import * as attributes from './attributes';
+
+/** Describes query params used with audit log query endpoint */
+export const auditLoqQuery = Type.Object({
+  user: Type.Optional(Type.String({
+    title: 'filter events by username',
+  })),
+  ip: Type.Optional(Type.String({
+    title: 'filter events by IP address',
+  })),
+  action: Type.Optional(
+    Type.Array( // TODO: Requires Ajv to be in coerce array mode
+      attributes.auditEventAction,
+      {
+        title: 'filter events by actions',
+      },
+    ),
+  ),
```

Comment:
Comes from a query string so there's no sense in making it an array. Don't know if we can assign a title this way, though.
```suggestion
  action: Type.Optional(attributes.auditEventAction),
```

Reply:
That's true, as we assume to get only one action to filter with. However, parsing query string attributes is a little ambiguous business: what to do when somebody makes a request with `action` defined twice? There is no universal agreement on which one of them to choose. To make the API more user-friendly, both of them should be considered as valid action filters, and that's why `action` is defined here as an array. Naturally, both of the requested action types should be included in the query result.